### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by pochtalon**
Finished tests

### CODE REVIEW

File: src/test/java/mate/intro/controller/AttachmentControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package mate.intro.controller;
+
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class AttachmentControllerTest {
+
+}
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/AttachmentControllerTest.java]: For what you need it?

File: src/test/resources/application.properties [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
-spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.datasource.url=jdbc:tc:mysql:8:///task_management
+spring.datasource.username=test
+spring.datasource.password=test
+
+jwt.expiration=3000000
+jwt.secret=ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn
+
+spring.config.import=optional:file:.env[.properties]
+dropbox.token=${TOKEN}
```
Comment (Sanderfuck) [inline src/test/resources/application.properties:9]: Is it possible to run tests without init this propertie?

File: src/test/java/mate/intro/controller/AuthControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        UserLoginRequestDto requestDto = new UserLoginRequestDto()
+                .setEmail(EMAIL)
+                .setPassword(PASSWORD);
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(post("/api/auth/login")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/AuthControllerTest.java]: Where use result variable?

File: src/test/java/mate/intro/controller/AuthControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    @Test
+    @DisplayName("Login invalid user")
+    public void login_InvalidLoginRequestDto_Status401() throws Exception {
+        UserLoginRequestDto requestDto = new UserLoginRequestDto()
+                .setEmail(EMAIL)
+                .setPassword(PASSWORD);
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(post("/api/auth/login")
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/AuthControllerTest.java]: Same

File: src/test/java/mate/intro/controller/LabelControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "tasks");
+        assertEquals(3, actual.getTasks().size());
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete label by label id")
+    public void deleteLabel_LabelId_Success() throws Exception {
+        MvcResult delete = mockMvc.perform(delete("/api/labels/" + LABEL_ID_1)
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/LabelControllerTest.java]: Delete variable, do you use it?

File: src/test/java/mate/intro/controller/ProjectControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+        MvcResult result1 = mockMvc.perform(get("/api/projects/" + PROJECT_ID_1)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        ProjectDto actual_first = objectMapper
+                .readValue(result1.getResponse().getContentAsString(), ProjectDto.class);
+        assertNotNull(actual_first);
+        assertEquals(NAME_FIRST, actual_first.getName());
+
+        MvcResult result12 = mockMvc.perform(get("/api/projects/" + PROJECT_ID_2)
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/ProjectControllerTest.java]: Why result12?

File: src/test/java/mate/intro/controller/TaskControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+    private static final String ADMIN_NICK = "admin";
+    private static final Long ASSIGNEE_ID = 2L;
+    private static final String ASSIGNEE_NICK = "mnemonic";
+    private static final String UP_NAME = "New name";
+    private static final String UP_DESC = "Description";
+    private static final String UP_PRIORITY = "HIGH";
+    private static final String UP_STATUS = "IN_PROGRESS";
+    private static final String UP_DATE = "2025-01-02";
+    private static final Long UP_ASSIGNEE = 1L;
+
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/TaskControllerTest.java:71]:
```suggestion
```

File: src/test/java/mate/intro/controller/TaskControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete task by task id")
+    public void deleteTask_TaskId_Success() throws Exception {
+        MvcResult delete = mockMvc.perform(delete("/api/tasks/" + USER_TASK_ID)
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/controller/TaskControllerTest.java]: delete variable

File: src/test/java/mate/intro/service/AttachmentServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class AttachmentServiceImplTest {
+    @InjectMocks
+    AttachmentServiceImpl attachmentService;
```
Comment (Sanderfuck) [inline src/test/java/mate/intro/service/AttachmentServiceImplTest.java]: Place after Mocks

**Review by Sanderfuck — CHANGES_REQUESTED** [review]
image
Committed code

File: src/test/java/mate/intro/service/AttachmentServiceImplTest.java (hunk 9 above)
Comment (pochtalon) [reply src/test/java/mate/intro/service/AttachmentServiceImplTest.java]: Do you mean, I should replace fild with annotation @InjectMocks after fields with annotation @Mock in each service test?

**Pull request merged**